File: RentIT/src/main/java/com/rentit/model/Product.java
package com.rentit.model;

import com.rentit.model.enums.ProductStatus;
import lombok.*;

import java.time.LocalDate;
import java.util.List;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
//TODO might need to convert to DTO class
public class Product {
    private int id;
    private String name;
    private String description;
    private float dayPrice;
    private float weekPrice;
    private float monthPrice;
    private float deposit;
    private String city;
    private float productValue;
    private int minLeasePeriod;
    private String category;
    private List<String> tags;
    private List<Image> images;
    private ProductStatus status;
    private LocalDate rentedUntil;
    private int userId;
}

File: RentIT-Frontend/src/model/product.ts
import {Image} from "src/model/image";
import {ProductStatus} from "src/model/productStatus";
import {ReviewsOverview} from "src/model/reviewsOverview";

export interface Product {
  id?: number;
  name: string;
  reviewsOverview?: ReviewsOverview;
  description?: string;
  dayPrice: number;
  weekPrice: number;
  monthPrice: number;
  deposit: number;
  city: string;
  productValue: number;
  minLeasePeriod?: number | null;
  category: string;
  tags: string[];
  images: Image[];
  status: ProductStatus;
  rentedUntil?: string | null;
}
